frontend/src/pages/HomePage/HomePage.tsx

```diff
@@ -1,9 +1,31 @@
-import React from "react";
-import "./HomePage.css";
+import React, { useEffect, useState } from "react";
 import "bootstrap/dist/css/bootstrap.min.css";
+import { useNavigate } from "react-router-dom";
+import { QuestionEditor } from "../../components/QuestionEditor/QuestionEditor.component";
+import { SolvedTable } from "../../components/SolvedTable/SolvedTable.component";
+import { sampleSolvedQuestions } from "../../data/sampleqn";
 
 function HomePage() {
-  return <></>;
+  const [userId, setUserId] = useState(null);
+
+  useEffect(() => {
+    // Fetch the user ID dynamically from your API or data source
+    fetch("http://localhost:8000/api/get-user-id")
+      .then((response) => response.json())
+      .then((data) => {
+        setUserId(data.userId);
```

Comment:
define a function in `/src/api` to hit the endpoint. And use `userServiceClient` instead of the `fetch` library

Reply:
Defined functions in user.ts under [Added user.ts to fetch userId and SolvedQuestions and changes to hompage](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g10/pull/43/commits/aa3c823c21fa81988a8977712842575ec7e8f19c)